DataFormats/ParticleFlowReco/interface/PFRecHit.h

```diff
@@ -7,20 +7,16 @@
 #include <iostream>
 
 #include "DataFormats/Math/interface/Point3D.h"
-#include "Rtypes.h" 
 #include "DataFormats/Math/interface/Vector3D.h"
 #include "Math/GenVector/PositionVector3D.h"
 #include "DataFormats/ParticleFlowReco/interface/PFLayer.h"
 #include "DataFormats/ParticleFlowReco/interface/PFRecHitFwd.h"
 
-//C decide what is the default rechit index. 
-//C maybe 0 ? -> compression 
-//C then the position is index-1. 
-//C provide a helper class to access the rechit. 
-
 #include "DataFormats/CaloRecHit/interface/CaloRecHit.h"
 #include "DataFormats/Common/interface/RefToBase.h"
 
+#include "Geometry/CaloGeometry/interface/CaloCellGeometry.h"
```

Comment:
The dependencies on Geometry are
```
cdj@cdj> git grep 'include "Geometry/' | grep DataFormats
DataFormats/TrackerRecHit2D/interface/BaseTrackerRecHit.h:#include "Geometry/CommonDetUnit/interface/TrackerGeomDet.h"
DataFormats/TrackerRecHit2D/interface/TrackerSingleRecHit.h:#include "Geometry/CommonDetUnit/interface/GeomDetUnit.h"
DataFormats/TrackerRecHit2D/interface/TrackingRecHitLessFromGlobalPosition.h:#include "Geometry/CommonDetUnit/interface/GeomDet.h"
DataFormats/TrackerRecHit2D/interface/TrackingRecHitLessFromGlobalPosition.h:#include "Geometry/TrackerGeometryBuilder/interface/GeomDetLess.h"
DataFormats/TrackerRecHit2D/interface/TrackingRecHitLessFromGlobalPosition.h:#include "Geometry/CommonDetUnit/interface/TrackingGeometry.h"
DataFormats/TrackerRecHit2D/src/ProjectedSiStripRecHit2D.cc:#include "Geometry/TrackerGeometryBuilder/interface/GluedGeomDet.h"
DataFormats/TrackingRecHit/interface/TrackingRecHit.h:#include "Geometry/CommonDetUnit/interface/GeomDet.h"
DataFormats/TrackingRecHit/src/TrackingRecHit.cc:#include "Geometry/CommonDetUnit/interface/GeomDetUnit.h"
```
Where only DataFormats/TrackingRecHit actually links to a Geometry library
https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_X/DataFormats/TrackingRecHit/BuildFile.xml

And Geometry/CommonDetUnit only depends on items in DataFormats and FWCore/Utilities
https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_X/Geometry/CommonDetUnit/BuildFile.xml

The only outlier is Geometry/TrackerGeometryBuilder/interface/GeomDetLess.h which from its code
https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_X/Geometry/TrackerGeometryBuilder/interface/GeomDetLess.h

shows that the class does not depend on anything in TrackerGeometryBuilder and therefore should be moved to another location.

So as far as I can tell the tracking code does not actually add any problematic dependencies on Geometry.